cmd/ipfs-cluster-ctl/graph.go

```diff
@@ -78,53 +84,138 @@ type dotWriter struct {
 	w        io.Writer
 	dotGraph dot.Graph
 
+	self             string
+	idToPeername     map[string]string
+	trustMap         map[string]bool
 	ipfsEdges        map[string][]peer.ID
 	clusterEdges     map[string][]peer.ID
 	clusterIpfsEdges map[string]peer.ID
 }
 
+func (dW *dotWriter) addSubGraph(sGraph dot.Graph, rank string) {
+	sGraph.IsSubGraph = true
+	sGraph.Rank = rank
+	dW.dotGraph.AddSubGraph(&sGraph)
+}
+
 // writes nodes to dot file output and creates and stores an ordering over nodes
-func (dW *dotWriter) addNode(id string, nT nodeType) error {
-	var node dot.VertexDescription
-	pid, _ := peer.IDB58Decode(id)
-	node.Label = pid.ShortString()
+func (dW *dotWriter) addNode(graph *dot.Graph, id string, nT nodeType) error {
+	node := dot.NewVertexDescription("")
+	node.Group = id
+	node.ColorScheme = "brbg11"
+	node.FontName = "Ariel"
+	node.Shape = "ellipse"
+	node.Style = "filled"
 	switch nT {
+	case tSelfCluster:
+		node.Label = label(dW.idToPeername[id], shorten(id))
+		node.ID = fmt.Sprintf("C%d", len(dW.clusterNodes))
+		node.Color = "11"
+		node.Peripheries = 2
+		node.FontColor = "6"
+		dW.clusterNodes[id] = &node
+	case tTrustedCluster:
+		node.Label = label(dW.idToPeername[id], shorten(id))
+		node.ID = fmt.Sprintf("T%d", len(dW.clusterNodes))
+		node.Color = "11"
+		node.FontColor = "6"
+		dW.clusterNodes[id] = &node
 	case tCluster:
+		node.Label = label(dW.idToPeername[id], shorten(id))
 		node.ID = fmt.Sprintf("C%d", len(dW.clusterNodes))
-		node.Color = "blue2"
+		node.Color = "9"
+		node.FontColor = "6"
 		dW.clusterNodes[id] = &node
-	case tIpfs:
+	case tIPFS:
+		node.ID = fmt.Sprintf("I%d", len(dW.ipfsNodes))
+		node.Shape = "box"
+		ipfsID, ok := dW.clusterIpfsEdges[id]
+		if !ok {
+			node.Label = label("IPFS", "Errored")
+			node.ColorScheme = "X11"
```

Comment:
How about limiting to colors in the `rdylbu9` color scheme

Reply:
(and only using colors from that scheme, for all things)

Reply:
It doesn't have white color. We would have to rely on light yellow or light blue for for fonts in darker ellipse/boxes, which I think would not be as good as white.